backEnd/src/users/users.controller.ts

```diff
@@ -0,0 +1,99 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete, HttpStatus } from '@nestjs/common';
+import { UsersService } from './users.service';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { ApiBody, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { UserDto } from './dto/user.dto';
+import { Roles } from 'decorators/roles.decorator';
+import { Type_user } from '@prisma/client';
+import { Public } from 'decorators/public.decorator';
+
+
+@ApiTags('users')
+@Controller('users')
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
+  @ApiOperation({summary: 'Create a new user!'})
+  @ApiBody({
+    type: CreateUserDto,
+    description: 'Create a new user',
+    required : true,
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'The user has been successfully created.',
+  })
+  @ApiResponse({
+    status: HttpStatus.UNPROCESSABLE_ENTITY,
+    description: 'The email is already being used!'
+  })
+  @ApiResponse({
+    status: HttpStatus.UNPROCESSABLE_ENTITY,
+    description: 'The user name is already being used!'
+  })
+  @Public()
```

Comment:
As we have a type in DTO, we can't have this route as public because everyone can create an admin account

Reply:
I remove the property type from create-user-dto